packages/components/src/dropdown-menu-v2-ariakit/types.ts

```diff
@@ -0,0 +1,178 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type * as Ariakit from '@ariakit/react';
+import type { Placement } from '@floating-ui/react-dom';
+
+export interface DropdownMenuContext {
+	store: Ariakit.MenuStore;
+	variant?: 'toolbar';
+}
+
+export interface DropdownMenuProps {
+	/**
+	 * The trigger button.
+	 */
+	trigger: React.ReactElement;
+	/**
+	 * The contents of the dropdown
+	 */
+	children?: React.ReactNode;
+	/**
+	 * The open state of the dropdown menu when it is initially rendered. Use when
+	 * not wanting to control its open state.
+	 */
+	defaultOpen?: boolean;
+	/**
+	 * The controlled open state of the dropdown menu. Must be used in conjunction
+	 * with `onOpenChange`.
+	 */
+	open?: boolean;
+	/**
+	 * Event handler called when the open state of the dropdown menu changes.
+	 */
+	onOpenChange?: ( open: boolean ) => void;
+	/**
+	 * The modality of the dropdown menu. When set to true, interaction with
+	 * outside elements will be disabled and only menu content will be visible to
+	 * screen readers.
+	 *
+	 * @default true
+	 */
+	modal?: boolean;
+	/**
+	 * The placement of the dropdown menu popover.
+	 *
+	 * @default 'bottom-start' for root-level menus, 'right-start' for nested menus
+	 */
+	placement?: Placement;
+	/**
+	 * The distance between the popover and the anchor element.
+	 *
+	 * @default 8 for root-level menus, 16 for nested menus
+	 */
+	gutter?: number;
+	/**
+	 * The skidding of the popover along the anchor element. Can be set to
+	 * negative values to make the popover shift to the opposite side.
+	 *
+	 * @default 0 for root-level menus, -8 for nested menus
+	 */
+	shift?: number;
+	/**
+	 * Determines whether the menu popover will be hidden when the user presses
+	 * the Escape key.
+	 *
+	 * @default true
+	 */
+	hideOnEscape?:
+		| boolean
+		| ( (
+				event: KeyboardEvent | React.KeyboardEvent< Element >
+		  ) => boolean );
+}
+
+export interface DropdownMenuGroupProps {
+	/**
+	 * The contents of the group
+	 */
+	children: React.ReactNode;
+}
+
+export interface DropdownMenuGroupLabelProps {
+	/**
+	 * The contents of the label
+	 */
+	children: React.ReactNode;
+}
+
+export interface DropdownMenuItemProps {
+	/**
+	 * The contents of the label
+	 */
```

Comment:
Nit: At first glance, this seemed like a copy-and-paste error from above. Could it be clearer as to what the contents are? Also, should all these comments end with a period?

Reply:
Thank you for spotting this! I went ahead and did a generic round of comment polishing in https://github.com/WordPress/gutenberg/pull/54939/commits/6bac80ccac987676185d36dd2098baf3dbefa9e9.

> Could it be clearer as to what the contents are?

We usually keep `children` prop descriptions generic, did you have anything specific in mind?